File: 2453-destroy-sequential-targets/2453-destroy-sequential-targets.py
class Solution:
    def destroyTargets(self, nums: List[int], space: int) -> int:
        tracker = {}
        minVal = float('inf')
        
        for n in nums:
            rem = n % space
            if not rem in tracker:
                tracker[rem] = 0
            tracker[rem] += 1
        maxTargets = max(tracker.values())
        
        for n in nums:
            rem = n % space
            if tracker[rem] == maxTargets and n < minVal:
                minVal = n
        
        return minVal

File: 2453-destroy-sequential-targets/2453-destroy-sequential-targets.js
/**
 * @param {number[]} nums
 * @param {number} space
 * @return {number}
 */
var destroyTargets = function(nums, space) {
    const tracker = {};
    for (let n of nums) {
        const rem = n % space;
        if (!(rem in tracker)) {
            tracker[rem] = 0;
        }
        tracker[rem]++;
    };
    let maxTargets = Math.max(...Object.values(tracker));
    
    let minVal = Infinity;
    for (let n of nums) {
        const rem = n % space;
        if (tracker[rem] === maxTargets && n < minVal) {
            minVal = n;
        }
    };
    
    return minVal;    
};